hs_core/tests/api/rest/test_zip_folder.py

```diff
@@ -0,0 +1,87 @@
+
+        self.tmp_dir = tempfile.mkdtemp()
+
+        # Make a text file
+        self.txt_file_name = 'text.txt'
+        self.txt_file_path = os.path.join(self.tmp_dir, self.txt_file_name)
+        txt = open(self.txt_file_path, 'w')
+        txt.write("Hello World\n")
+        txt.close()
+
+        self.raster_file_name = 'cea.tif'
+        self.raster_file_path = 'hs_core/tests/data/cea.tif'
+
+        self.rtype = 'GenericResource'
+        self.title = 'My Test resource'
+        res = resource.create_resource(self.rtype,
+                                       self.user,
+                                       self.title,
+                                       unpack_file=False)
+
+        self.pid = res.short_id
+        self.resources_to_delete.append(self.pid)
+
+        # create a folder 'foo'
+        url = str.format('/hsapi/resource/{}/folders/foo/', self.pid)
+        self.client.put(url, {})
+
+        # put a file 'test.txt' into folder 'foo'
+        url2 = str.format('/hsapi/resource/{}/files/foo/', self.pid)
+        params = {'file': ('text.txt',
+                           open(self.txt_file_path, 'rb'),
+                           'text/plain')}
+        self.client.post(url2, params)
+
+        # put a file 'cea.tif' into folder 'foo'
+        url3 = str.format('/hsapi/resource/{}/files/foo/', self.pid)
+        params = {'file': (self.raster_file_name,
+                           open(self.raster_file_path, 'rb'),
+                           'image/tiff')}
+        self.client.post(url3, params)
+
+    def test_zip_folder_bad_requests(self):
+        zip_url = "/hsapi/resource/%s/functions/zip/" % self.pid
+
+        response_no_path = self.client.post(zip_url, {
+            "output_zip_file_name": "test.zip"
+        }, format="json")
+        response_empty_path = self.client.post(zip_url, {
+            "output_zip_file_name": "test.zip"
+        }, format="json")
+        response_no_fname = self.client.post(zip_url, {
+            "output_zip_file_name": " ",
+            "input_coll_path": "/files/foo"
+        }, format="json")
+        response_empty_fname = self.client.post(zip_url, {
+            "output_zip_file_name": "test.zip",
+            "input_coll_path": " "
+        }, format="json")
+
+        self.assertEqual(response_no_path.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response_empty_path.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response_no_fname.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response_empty_fname.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_zip_folder(self):
+        zip_url = "/hsapi/resource/%s/functions/zip/" % self.pid
+        response = self.client.post(zip_url, {
+            "input_coll_path": "data/contents/foo",
+            "output_zip_file_name": "data/contents/test.zip",
+            "remove_original_after_zip": False
+        }, format="json")

```

Comment:
@aphelionz output_zip_file_name should be only 'test.zip', and other parameters look correct to me. If you set input_coll_path this way, I assume you already have ```foo``` as a folder under /data/contents and the whole foo folder should be zipped into test.zip. See if that works.